csharp/src/Ice/Internal/IdleTimeoutTransceiverDecorator.cs

```diff
@@ -61,17 +61,8 @@ public void finishRead(Buffer buf)
 
     public ConnectionInfo getInfo() => _decoratee.getInfo();
 
-    public int initialize(Buffer readBuffer, Buffer writeBuffer, ref bool hasMoreData)
-    {
-        int op = _decoratee.initialize(readBuffer, writeBuffer, ref hasMoreData);
-
-        if (op == SocketOperation.None) // connected
-        {
-            rescheduleWriteTimer();
```

Comment:
Bug 1: 
Scheduling the first "send heartbeat" in initialize is too early for a client connection that waits for a while to get the ValidateConnection message from the server. We can end up in the situation where the client connection sends its first heartbeat while the Connection is not Active yet, and all the send heartbeats are then lost if the client doesn't write anything else.